### PR TITLE
issue #25680 - add commercialcore if necessary and make createpkgschema idempotent

### PR DESCRIPTION
**Pull request by gilmoskowitz**

### CODE REVIEW

File: foundation-database/public/functions/createpkgschema.sql [hunk 1]
```diff
@@ -29,51 +27,49 @@ BEGIN
   END IF;
 
   FOR i IN ARRAY_LOWER(_tabs,1)..ARRAY_UPPER(_tabs,1) LOOP
-    _pkgtab := pname || '.pkg' || _tabs[i];
+    _pkgtab := 'pkg' || _tabs[i];
 
     IF NOT EXISTS(SELECT oid
                   FROM pg_class
                   WHERE ((relname=_pkgtab)
```
Comment (gilmoskowitz) [inline foundation-database/public/functions/createpkgschema.sql]: The bug that made `createpkgschema` non-idempotent was here in the `not exists(select...)`. There is no row in `pg_class` for a table with a name literally containing the schema name and a `.`. To fix it, though, all of the queries that referred to the `_pkgtab` variable needed to be rewritten. Hopefully the new version of the function is more readable.

File: scripts/lib/build_database.js [hunk 2]
```diff
@@ -50,7 +50,25 @@ var  async = require('async'),
...
+      ;
+
+      // bug 25680 - we added dependencies on a new commercialcore extension.
+      // make sure it's registered for installation if necessary
+      if (commercialPos >= 0 && commercialcorePos < 0) {
+        console.log(extensions);
+        commercialcorePath = commercialcorePath.replace(commercialRegexp, "commercialcore");
+        extensions.splice(commercialPos, 0, commercialcorePath);
+        console.log(extensions);
+      }
```
Comment (gilmoskowitz) [inline scripts/lib/build_database.js]: This feels like an incredible hack. If the list of schemas includes `inventory`, make sure that `commercialcore` is in the list before it. I hoped that build_app would automatically handle prerequisites. All I see, though, is injection of a line in the resulting .sql that causes an error if the prereq is not met.

**Comment by Panduk** [comment]
Performed UAT: Issue is fixed.

**Comment by gilmoskowitz** [comment]
pclark says +1

**Pull request merged**